src/tracer.js

```diff
@@ -228,6 +228,15 @@ export default class Tracer {
             ctx.parentId = null;
             ctx.flags = flags;
         } else {
+            if (!parent.samplingFinalized) {
+                if (this._sampler.isSampled(operationName, internalTags)) {
+                   parent._flags |= constants.SAMPLED_MASK
+                } else {
+                    parent._flags &= ~constants.SAMPLED_MASK
+                }
+                parent.finalizeSampling()
```

Comment:
we do not have access to parent span when creating a child span, we only have access to its context